### ISSUE TITLE
[Feature Request] Map the Middle Mouse Button (Click Wheel) to open the Workspaces overview.

### ISSUE THREAD

**Issue by adamovera** [pop-os/gnome-shell-extension-pop-cosmic#76]
Please map the middle mouse button (click wheel) to activate the Workspaces overview. It would make so much sense since the Scroll Wheel already switches between workspaces once the Workspaces overview is active. There doesn't appear to be any consistent system-wide behavior for the middle mouse button anyway, nor any options in Settings to modify what it does. I feel like mouse users would really appreciate this shortcut. Thank you!

**Comment by blakeridgway** [comment]
Middle Mouse button has a lot of functionality in other programs. For example, you can close web browser tabs with middle Mouse button

**Comment by adamovera** [comment]
> Middle Mouse button has a lot of functionality in other programs. For example, you can close web browser tabs with middle Mouse button

Yeah, and pastes, and opens a new instance of items in the Dock, and opens new tabs in Nautilus, and who knows what else. But that's the problem, it does so many things that no one knows what it does, and it often just goes unused. Being able to configure what it does system-wide should be an option in Settings.

**Comment by blakeridgway** [comment]
While that would be nice, but there are so many other applications in play that you cannot control ALL of their keybinds.

**Comment by adamovera** [comment]
> While that would be nice, but there are so many other applications in play that you cannot control ALL of their keybinds.

Right, but I'd want to override all that with a single system-wide middle mouse behavior. Having it do all that stuff makes it too niche to be very useful. And it's confusing, every other shortcut does one thing - between the two of us we can't list everything this button does. That said, you could default to the current per-app behavior, but provide a handful of logical, system-wide, mouse-driven options for things like open Applications, open Workspaces, Paste, Show Desktop, Maximize Window, etc. in a Middle Button entry under Settings/Mouse & Touchpad/General.

**Comment by Avendattor** [comment]
I think that default MMB action is useful in a lot of apps. All my ~~homies~~ friends use it and no one was thinking about redefining it.

> override all that with a single system-wide middle mouse behavior

IMHO, it can be useful for some amount of users. But for a very small amount.

**Comment by spxak1** [comment]
The middle mouse button is important in Linux. You can use any standard method to map it to any key combination, but Pop should not do this. Thanks.

**Comment by adamovera** [comment]
> The middle mouse button is important in Linux. You can use any standard method to map it to any key combination, but Pop should not do this. Thanks.

The "standard method" should be right in Settings/Mouse & Trackpad/General since every mouse has a click wheel.

**Comment by cbozic** [comment]
First, I'm a long time linux user but fairly new to PopOS and this is one of the best linux experiences I've ever had!  It's opened my eyes to all the amazing hardware work of System76 too!

I want to join the conversation here because I also would like to **have the option** to use the middle mouse button to activate workspaces (or maybe someone else would like it to open the application launcher).  I love the speed it gives me switching between apps in non-tile mode.  I'm not suggesting it should be a default setting because I agree that the middle click is useful in certain apps for certain users.  However, for users like @adamovera  and me who don't care about using the specific behavior in each application, it would really speed up our workflows to be able to assign extra mouse buttons to open workspaces or the app launcher.

I could stop here but I'll go on to say that I have a pretty decent mouse with several extra buttons beyond just the usual three.  So, I would love it if there was a way to assign really any of those extra buttons to do common desktop functions.

Having said all this, I'm going to go [read this article and find out how to accomplish my goal](https://www.linuxuprising.com/2019/11/how-to-bind-mouse-buttons-to-keyboard.html) without this functionality directly in the OS.  

Thanks to everyone for all the great work on PopOS!

**Comment by adamovera** [comment]
> First, I'm a long time linux user but fairly new to PopOS and this is one of the best linux experiences I've ever had! It's opened my eyes to all the amazing hardware work of System76 too!
> 
> I want to join the conversation here because I also would like to **have the option** to use the middle mouse button to activate workspaces (or maybe someone else would like it to open the application launcher). I love the speed it gives me switching between apps in non-tile mode. I'm not suggesting it should be a default setting because I agree that the middle click is useful in certain apps for certain users. However, for users like @adamovera and me who don't care about using the specific behavior in each application, it would really speed up our workflows to be able to assign extra mouse buttons to open workspaces or the app launcher.
> 
> I could stop here but I'll go on to say that I have a pretty decent mouse with several extra buttons beyond just the usual three. So, I would love it if there was a way to assign really any of those extra buttons to do common desktop functions.
> 
> Having said all this, I'm going to go [read this article and find out how to accomplish my goal](https://www.linuxuprising.com/2019/11/how-to-bind-mouse-buttons-to-keyboard.html) without this functionality directly in the OS.
> 
> Thanks to everyone for all the great work on PopOS!

Thanks! I've been using [Key Mapper](https://github.com/sezanzeb/input-remapper) to configure my mouse buttons.

**Comment by antic1pate** [comment]
I just installed popOS! on my old 2009 imac. I was also looking for this functionality.
